### PR TITLE
add basic tests without SMP support

### PR DESCRIPTION
**Pull request by stlankes**

### CODE REVIEW

File: .github/workflows/nightly_nosmp.yml [hunk 1]
```diff
@@ -0,0 +1,119 @@
+name: Tests with SMP support (nightly)
```
Comment (jbreitbart) [inline .github/workflows/nightly_nosmp.yml:1]: Either the filename is wrong or the name

File: .github/workflows/nightly_nosmp.yml [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        with:
+          path: 'libhermit-rs'
+      - name: Check Cargo availability
+        run: cargo --version
+      - name: Cargo Test libhermit-rs (Unittests on Host)
+        run: cargo test --lib --target x86_64-unknown-linux-gnu
+        working-directory: libhermit-rs
+        if: ${{ matrix.os == 'ubuntu-latest' }}
+      - name: Install qemu/nasm (apt)
+        run: sudo apt-get update --fix-missing && sudo apt-get install qemu-system-x86 nasm
```
Comment (jbreitbart) [inline .github/workflows/nightly_nosmp.yml:67]: Is nasm still required?

File: .github/workflows/nightly_nosmp.yml [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+      - staging
+      - trying
+
+jobs:
+  build:
+    name: Tests without SMP support
+    runs-on: ${{ matrix.os }}
+    # these tests based on the nightly compiler, which can be broken
+    # consequently, we continue on an error
+    #continue-on-error: true
```
Comment (jbreitbart) [inline .github/workflows/nightly_nosmp.yml:17]: Should this be enabled? If not, can you remove it?

File: .github/workflows/nightly.yml [hunk 4]
```diff
@@ -90,10 +90,10 @@ jobs:
         if: ${{ matrix.os == 'windows-latest' }}
       - name: Building dev version
         run:
-          cargo build -Z build-std=std,core,alloc,panic_abort --target x86_64-unknown-hermit
+          cargo build -p rusty_demo -Z build-std=std,core,alloc,panic_abort --target x86_64-unknown-hermit
       - name: Building release version
         run:
-          cargo build -Z build-std=std,core,alloc,panic_abort --target x86_64-unknown-hermit --release
+          cargo build -p rusty_demo -Z build-std=std,core,alloc,panic_abort --target x86_64-unknown-hermit --release
```
Comment (jbreitbart) [inline .github/workflows/nightly.yml:96]: Why are you only building rusty_demo?

**Review by jbreitbart — CHANGES_REQUESTED** [review]

File: .github/workflows/nightly_nosmp.yml (hunk 1 above)
Comment (stlankes) [reply .github/workflows/nightly_nosmp.yml]: Oh, a typo... I fixed it.

File: .github/workflows/nightly_nosmp.yml (hunk 2 above)
Comment (stlankes) [reply .github/workflows/nightly_nosmp.yml:67]: `nasm` is required to build the boot loader, which we need to run the kernel within qemu. The unikernel doesn't depend on `nasm`.

File: .github/workflows/nightly_nosmp.yml (hunk 3 above)
Comment (stlankes) [reply .github/workflows/nightly_nosmp.yml]: I removed these lines.

File: .github/workflows/nightly.yml (hunk 4 above)
Comment (stlankes) [reply .github/workflows/nightly.yml:93]: It is currently the only test, which is used.

Comment (jbreitbart) [reply .github/workflows/nightly.yml:93]: That is only true in this PR. The one you just merged added a standalone pi sequential test

Comment (stlankes) [reply .github/workflows/nightly.yml:93]: But the standalone pi sequential test is also included in rusty_demo

**Comment by stlankes** [comment]
bors try

File: .github/workflows/nightly.yml (hunk 4 above)
Comment (jbreitbart) [reply .github/workflows/nightly.yml:93]: > But the standalone pi sequential test is also included in the package `rusty_demo`. Consequently, we build also the standalone version of the pi sequential test.

No, you don't. If I change something in the sequential pi test that does not compile, you will not notice it with this CI pipeline. Furthermore, if you change something in the demo and not adapt the sequential pi test, you may break the standalone test without realizing it. We can remove the standalone tests again, but even than I fail to see the benefit of that change. 🤷

**Pull request closed without merging**